worlds/kh1/Rules.py

```diff
@@ -0,0 +1,553 @@
+from BaseClasses import CollectionState, MultiWorld, LocationProgressType
+from .Locations import get_locations_by_category
+
+
+def has_x_worlds(state: CollectionState, player: int, num_of_worlds) -> bool:
+    return sum([1 for world in ["Wonderland", "Olympus Coliseum", "Deep Jungle",
+                                         "Agrabah", "Monstro", "Atlantica",
+                                         "Halloween Town", "Neverland", "Hollow Bastion",
+                                         "End of the World"] if has_item(state, player, world)]) >= num_of_worlds
```

Comment:
I'd say to use `state.count_group` here, but if someone uses `start_inventory` to give themselves a world, it could make this give the wrong number. If https://github.com/ArchipelagoMW/Archipelago/pull/3192 gets merged, I recommend using `state.count_group_exclusive` afterwards.